huaweicloud/services/obs/resource_huaweicloud_obs_bucket.go

```diff
@@ -737,14 +742,22 @@ func resourceObsBucketEncryptionUpdate(config *config.Config, obsClient *obs.Obs
 	if d.Get("encryption").(bool) {
 		input := &obs.SetBucketEncryptionInput{}
 		input.Bucket = bucket
-		input.SSEAlgorithm = obs.DEFAULT_SSE_KMS_ENCRYPTION_OBS
-		input.KMSMasterKeyID = d.Get("kms_key_id").(string)
 
-		if _, ok := d.GetOk("kms_key_id"); ok {
-			if v, ok := d.GetOk("kms_key_project_id"); ok {
-				input.ProjectID = v.(string)
-			} else {
-				input.ProjectID = config.GetProjectID(config.GetRegion(d))
+		if v, ok := d.GetOk("sse_algorithm"); ok {
+			input.SSEAlgorithm = v.(string)
+		} else {
+			input.SSEAlgorithm = obs.DEFAULT_SSE_KMS_ENCRYPTION_OBS
+		}
+
+		if input.SSEAlgorithm == obs.DEFAULT_SSE_KMS_ENCRYPTION_OBS {
+			input.KMSMasterKeyID = d.Get("kms_key_id").(string)
+
+			if _, ok := d.GetOk("kms_key_id"); ok {
```

Comment:
this will make the code more clear
```suggestion
			if raw, ok := d.GetOk("kms_key_id"); ok {
				input.KMSMasterKeyID = raw.(string)
```